### DISCUSSION TITLE
Request: New Features

### DISCUSSION THREAD

**Discussion by rumboalla** [rumboalla/apkupdater#393] [General]
Compiled this list from previously open issues. Listed in order of importance, more or less.

- Progress bar for downloads.
- Wifi only update check.
- Control cache behaviour and size.
- Sorting/Ordering of updates.
- Download notifications.
- Add link to release page.
- Darker theme #000000
- Ignore toggle for non-release apps.
- Ignore all option.
- Merge sources.
- Shizuku support.
- Whitelist instead of blacklist.
- Collapsed ignored apps.
- Download APK only option.
- Fast scroller

If any feature needs more discussion, feel free to open a new thread.

**Comment by VasilisPngs** [comment]
Add the dpi option as well.

**Comment by warmoans** [comment]
If you could add pull down to refresh on the updates screen that would be cool. It currently only works on the installed apps screen. 

Edit: it looks like pull to refresh doesn't work on the lower half of the screen. Only on the top half.

**Reply by warmoans** [reply]
I've attached a video showing this. 

Edit: Removed link as issue is fixed.

**Reply by rumboalla** [reply]
This will be fixed on next version

**Reply by warmoans** [reply]
It's fixed now so I'm taking the video down from my mega. Thanks for fixing it.

**Comment by mefinity** [comment]
I have a request :3
A possibility to use [Shizuku](https://shizuku.rikka.app/) for silent installs

**Reply by rumboalla** [reply]
Added this to the list. 
If you are on Android 12+ silent installs are already supported.

**Comment by daffamumtaz2361** [comment]
I'd like to have an option to download specific/older version. Not only install but also an option to download the apk file.

Also it would be nice to be able to choose which source to download from

**Comment by AnonymousWP** [comment]
Why actually feature requests in a discussion? That's not really optimal imo, because:

1. Now users who want to contribute can't link it to an issue (because there's none). Theyll have to find the discussion first (which is inefficient as issues are the way-to when looking for pending bugs or feature requests)
2. It's against good Git practices
3. You as a maintainer would have to keep updating the list you created at the beginning of this discussion

**Reply by rumboalla** [reply]
I find it easier this way. The important issues are in issues and don't get buried by tons of stuff and the feature requests are curated in a handful of discussions.
It is OK to open issues for features that are being implemented and link to it.

**Reply by AnonymousWP** [reply]
But what then happens is that the issue of the feature request is closed, and thus you can't link to it anymore in a PR (or well, you can, but what's the point as it's already closed?).

**Comment by Pantyhose-X** [comment]
GET
Max Connections per Download 64  Download APK

**Comment by Helpful-Neko** [comment]
Apologies for posting this in the wrong section first. I forgot to select the proper form. Again though thank you for the work you've put into this!

Ideas to ponder that I think would make good features and benefit the app for a smoother experience 

(1) Ability to see updates source and applications source. For example tapping on the application to see it's source and the source for a update.

Theses features would make it a little bit more convenient to see whats in the new update (instead of blindly updating it) and saves time in my opinion. Something equivalent to how you can pull up the application information on Droid-ify. Being able to go directly to the apps source [GitHub, APK mirror ect] alongside a list of the new features in the update you're installing/viewing.  Maybe even a previous versions list for comparison 

(2) The ability to ignore updates (all new updates) for certain applicants alongside ignoring the current update (Already on your list)

(3) Being able to have Shizuku Installer as a 3rd installer option. I think it would be really beneficial.
(Already on your list)

(4) Having the ability to have some applications set up to automatically update while other's can be set to manually update. That would help with applications that are sometimes know to be released with bugs/issues in their newer versions but allowing others to update automatically. 
Personally I'd love a feature like that. 

Reworded repost of #453 I posted in the wrong place accidentally. 

Edited for a typo and better clarification.

**Comment by mpc-v** [comment]
[Request] Compile-time option to disable the search tab

I'm trying to make my phone as distraction-free as possible while still keeping my needed apps up to date. The search feature makes it too easy to install distracting apps in moments of weakness. A compile-time option to disable search would greatly reduce my ability to install distracting apps. I am not asking you to provide builds with this option (though that would be really great).

**Comment by TotallyAvailable** [comment]
Might seem wild and far away from what you would want to invest time into so I'm not requesting it, just dropping it here as something I thought about.
- Icon Pack support 

I'll add two more
Given
> - Collapsed ignored apps. 

How about making it a toggle like 'hide system apps' and 'hide disabled'

Also an option in the Settings page for a  "one time full check" including Ignore Apps.

**Comment by sharjeel-butt** [comment]
Two small feautes for improved usability.
   - List view like what we have in v2.
   - Version upgrade info like we have in v2 (currently installed version -> Upgrade Version)

**Comment by imxade** [comment]
Update all <kbd>button</kbd>

**Comment by HazakuraID** [comment]
- Auto update check notify

**Comment by paulovbettio** [comment]
This app is would be really useful for people who are trying to make their smartphones "dumber" by removing all kinds of browsers, stores and social media apps, while maintaining some other important apps like banking and music ones.
However, requiring a browser to download the apps kills it for this specific use case. Implementing a **custom, embedded web-view to download the app** is a nice solution if direct downloads can't be implemented.

**Reply by TaivasJumala** [reply]
True bro, essentially needed for Android TV.

**Comment by 20171213** [comment]
apatch root has been authorized, but the program does not seem to recognize it, as shown by the fact that clicking the root installation switch does not show any effect.

**Comment by PeterKW** [comment]
Is it possible to have a way of viewing the download link of a suggested APK update?

**Reply by PeterKW** [reply]
https://github.com/rumboalla/apkupdater/discussions/482

**Comment by FavMaster98** [comment]
A feature that I would like to be added is the ability to add custom sources for tracking.

**Comment by xt1zer** [comment]
To watch an app's updates from the chosen source. From what I understand, in 3.0.3, watching an app's updates will look for all sources in the Updates screen. Given that a user can be experienced enough to decide which app store he'd like to look for updates for a specific app from, I think it would be very cool, and it'll only take one icon space on the Updates screen at most when an update is available.
In addition to the feature suggestion above, I also think that the "Update all" button would be relevant, but only show/enable it when the user has chosen one source for each watched app.

**Comment by nickk9** [comment]
Show updates from different sources grouped by the updating app. I.e. when there are updates for "Google drive" from apkpure and google play, only one "Google drive" should be listed and when you tap on it, it shows the table of sources where you can choose which to use for update.

**Comment by je-vv** [comment]
Just in case, not sure what happened, but without a new release for some time, now apkupdater is able to install software from apkpure again.  Not sure if this is just temporal, or what's going on, but it seems it's working back.

**Comment by ghost** [comment]
I have a couple of itches and I do have some time, so I am considering a fork to add some things. For one, I'm irritated about the reliability of downloads, so I am thinking about patching in something like OkHTTP or else looking at simply using wget or the libwget2. The other thing is I have a persistent threat, so with frequent resets of my android devices, it would be handy to cache verifiable, maybe xapk format, files for reinstallation. If so, maybe add an xapk installer, fairly trivial, simple format. I also have some gripes about UI, but it's android anyway, wouldn't it be cool if all the APIs in this app were exposed so you could also build a command line tool?

My several thoughts. I'm in a very good place for artisan pastries and premium coffee, so I might get to this myself this month.

I should mention that am something of a progress bar specialist.

**Comment by austin-agronick** [comment]
Adb can install apk files without user interaction on android 10. Is it possible for apkupdater to use that same method to add this feature for devices below android 12?

**Comment by Novaliss** [comment]
The option to un-ignore an ignored version, maybe even a list of 'ignores'.

❤️
Novaliss

**Comment by Tada82** [comment]
The ability to be able to see previously installed apps to be able to trouble shoot which app may be causing issues after update

**Comment by ciros88** [comment]
Ironfox (https://github.com/ironfox-oss/IronFox) support
